.github/workflows/maven.yaml

```diff
@@ -21,13 +21,31 @@ jobs:
         with:
           distribution: 'temurin'
           java-version: '17'
+
+      - name: Extract branch name
```

Comment:
we do the same thing in circleci and this fixes the build failure we have. to be fair, I am not yet 100% sure exactly why it happens to begin with, and of course I can't reproduce it locally (yet). in the meanwhile, this fixes the problem...